### PR TITLE
feature/repairswagger

### PR DESCRIPTION
**Pull request by sumitabarahmand**
## Description
SDK changes for the new Alert Repair admin operation
PR for the spec: https://github.com/Azure/azure-rest-api-specs/pull/4084
---

This checklist is used to make sure that common guidelines for a pull request are followed.
- [ ] Please add REST spec PR link to the SDK PR
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [ ] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [ ] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [ ] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [ ] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [ ] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

**Comment by sumitabarahmand** [comment]
@bganapa and @deathly809 
Spec PR link:
https://github.com/Azure/azure-rest-api-specs/pull/4084

**Comment by deathly809** [comment]
@sumitabarahmand After your spec PR has been merged please regenerate from merged spec.

File: src/AzureStack/Admin/InfrastructureInsightsAdmin/InfrastructureInsights.Admin/Microsoft.AzureStack.Management.InfrastructureInsights.Admin.csproj [hunk 1]
```diff
@@ -7,7 +7,7 @@
     <PropertyGroup>
         <PackageId>Microsoft.AzureStack.Management.InfrastructureInsights.Admin</PackageId>
         <Description>Provides developers with libraries for the health admin under AzureStack to manage health alerts.</Description>
-        <VersionPrefix>0.2.0-preview</VersionPrefix>
+        <VersionPrefix>0.3.0-preview</VersionPrefix>
         <AssemblyName>Microsoft.AzureStack.Management.InfrastructureInsights.Admin</AssemblyName>
         <PackageTags>Microsoft Azure Stack;InfrastructureInsights;REST HTTP client;azureostackfficial;netcore451511</PackageTags>
```
Comment (dsgouda) [inline src/AzureStack/Admin/InfrastructureInsightsAdmin/InfrastructureInsights.Admin/Microsoft.AzureStack.Management.InfrastructureInsights.Admin.csproj]: Please update `PackageReleaseNotes`

**Review by dsgouda — CHANGES_REQUESTED** [review]
When ready, please squash commits into a single commit

**Comment by bganapa** [comment]
@dsgouda GitHub has the option of squashing the commits when you are merging the PR. We are ok for you to do this. Pls lets us know if you still want us to manually squash the commits

**Comment by shahabhijeet** [comment]
@bganapa we lately started seeing issue with merges when we use squash/merge options, especially while trying to merge into other branches.
So we have now started being very particular on not having really long commit history. I understand 2-3 commits. But when you go beyond that, we ask RPs to clean up the commit history.
That is just something that helps us as well as you to have clean commit history, especially when we need to revert changes or when we have to create a branch out of a commit.
So help us to maintain the commit history in a clean state.

**Comment by bganapa** [comment]
Yes, we will squash the commits and update this PR once the specs PR is merged https://github.com/Azure/azure-rest-api-specs/pull/4084

**Comment by bganapa** [comment]
@sumitabarahmand could you please squash the commits.. 
Can you please add a release note saying the repair api is supported azurestack 1811 +

File: src/AzureStack/Admin/InfrastructureInsightsAdmin/InfrastructureInsights.Admin/Microsoft.AzureStack.Management.InfrastructureInsights.Admin.csproj [hunk 2]
```diff
@@ -7,14 +7,15 @@
...
         <PackageReleaseNotes>
             <![CDATA[
                 Preview release of the InfrastructureInsights administrator SDK which allows administrators  
                 1) View the health of their AzureStack stamp resources 
                 2) View and manage alerts
+                Repair API is supported in azurestack 1811+
```
Comment (dsgouda) [inline src/AzureStack/Admin/InfrastructureInsightsAdmin/InfrastructureInsights.Admin/Microsoft.AzureStack.Management.InfrastructureInsights.Admin.csproj]: nit: please stick to numbering scheme

File: src/AzureStack/Admin/InfrastructureInsightsAdmin/InfrastructureInsights.Admin/Generated/Models/Alert.cs [hunk 3]
```diff
@@ -72,7 +72,9 @@ public Alert()
...
         /// alert.</param>
-        public Alert(string id = default(string), string name = default(string), string type = default(string), IDictionary<string, string> tags = default(IDictionary<string, string>), string location = default(string), string closedTimestamp = default(string), string createdTimestamp = default(string), IList<IDictionary<string, string>> description = default(IList<IDictionary<string, string>>), string faultId = default(string), string alertId = default(string), string faultTypeId = default(string), string lastUpdatedTimestamp = default(string), IDictionary<string, string> alertProperties = default(IDictionary<string, string>), IList<IDictionary<string, string>> remediation = default(IList<IDictionary<string, string>>), string resourceRegistrationId = default(string), string resourceProviderRegistrationId = default(string), string severity = default(string), string state = default(string), string title = default(string), string impactedResourceId = default(string), string impactedResourceDisplayName = default(string), string closedByUserAlias = default(string))
+        /// <param name="hasValidRemediationAction">Indicates if the alert can
```
Comment (dsgouda) [inline src/AzureStack/Admin/InfrastructureInsightsAdmin/InfrastructureInsights.Admin/Generated/Models/Alert.cs]: Was this autogenerated or hand edited?

**Review by dsgouda — CHANGES_REQUESTED** [review]
Few minor comments

File: src/AzureStack/Admin/InfrastructureInsightsAdmin/InfrastructureInsights.Admin/Microsoft.AzureStack.Management.InfrastructureInsights.Admin.csproj (hunk 2 above)
Comment (deathly809) [reply src/AzureStack/Admin/InfrastructureInsightsAdmin/InfrastructureInsights.Admin/Microsoft.AzureStack.Management.InfrastructureInsights.Admin.csproj]: @dsgouda Should we only put in new things for release notes?

Comment (dsgouda) [reply src/AzureStack/Admin/InfrastructureInsightsAdmin/InfrastructureInsights.Admin/Microsoft.AzureStack.Management.InfrastructureInsights.Admin.csproj]: Ideally yes, that way users know what changed from the last version

Comment (sumitabarahmand) [reply src/AzureStack/Admin/InfrastructureInsightsAdmin/InfrastructureInsights.Admin/Microsoft.AzureStack.Management.InfrastructureInsights.Admin.csproj]: Just updated the PR.

**Review by dsgouda — APPROVED** [review]
LGTM

**Comment by dsgouda** [comment]
Will merge on CIs passing

File: src/AzureStack/Admin/InfrastructureInsightsAdmin/InfrastructureInsights.Admin/Microsoft.AzureStack.Management.InfrastructureInsights.Admin.csproj (hunk 1 above)
Comment (bganapa) [reply src/AzureStack/Admin/InfrastructureInsightsAdmin/InfrastructureInsights.Admin/Microsoft.AzureStack.Management.InfrastructureInsights.Admin.csproj]: please mention that repair api is supported only on AzureStack 1811 +

Comment (bganapa) [reply src/AzureStack/Admin/InfrastructureInsightsAdmin/InfrastructureInsights.Admin/Microsoft.AzureStack.Management.InfrastructureInsights.Admin.csproj]: @dsgouda can we add a checklist item in the PR saying 'I have updated relevant package release notes' ?

**Review by bganapa — APPROVED** [review]

**Pull request merged**